client/js/widgets/widget.js

```diff
@@ -311,18 +354,80 @@ export class Widget extends StateManaged {
     const routine = this.get(property) !== undefined ? this.get(property) : property;
 
     for(const original of routine) {
-      const a = JSON.parse(JSON.stringify(original));
+      let a = JSON.parse(JSON.stringify(original));
+      if(typeof a == 'object')
+        a = evaluateVariablesRecursively(a);
       var problems = [];
 
       if(this.get('debug')) console.log(`${this.id}: ${JSON.stringify(original)}`);
 
-      if(a.applyVariables) this.applyVariables(a, variables, problems);
-
       if(a.skip) {
         $('#debugButtonOutput').textContent += '\n\n\nOPERATION SKIPPED: \n' + JSON.stringify(a, null, '  ');
         continue;
       }
 
+      if(typeof a == 'string') {
+        const identifier = '(?:[a-zA-Z0-9_-]|\\\\u[0-9a-fA-F]{4})+';
```

Comment:
Identifier looks like a non-capturing group regular expression, but the others look like capturing groups. Also this expression allows identifiers to start with any character (or escape sequence): all numbers will match identifier.

Reply:
True and true. I don't see a problem though. I'm pretty sure the only way this would be ambiguous is if we had an operation name that only contains digits. If you can provide an example that doesn't match correctly, I'm very interested.